### ISSUE TITLE
missing proper routing

### ISSUE THREAD

**Issue by maximburdiean** [Alexpol19/AdminUi#1]
Не используется маршрутизация React для навигации. Вместо этого запоминается индекс выбранного раздела в Redux и отображается соответствующая страницу в зависимости от этого:
![image](https://user-images.githubusercontent.com/5040836/98915518-b1398100-24d2-11eb-837a-0426b3adc18a.png)

Из-за этого не работает браузерная навигация. URL при переходе на страницы остается неизменным и если нажать "Назад" в браузере, то пользователя просто выкидывает с сайта. Иконка для меню также определяется по индексу

**Comment by Alexpol19** [comment]
Согласен. Серьезный минус. Все-таки это страница а не просто Tab

**Comment by maximburdiean** [comment]
Попробуйте это исправить.

**Comment by Alexpol19** [comment]
Эту проблему решил с помощью react-router-dom.

**Comment by aandrey1993** [comment]
Работает всё правильно. Единственное что смущает - названия страниц из `aside.js` никак не связанно с маршрутизацией `Content.js`. Из-за этого название маршрутов повторяются при определении route, а также при формировании меню. Это не очень удобно поддерживать и увеличивает вероятность ошибок:

![изображение](https://user-images.githubusercontent.com/49781313/99110386-fd7be280-25f2-11eb-8d59-5b4d46dad94c.png)

![изображение](https://user-images.githubusercontent.com/49781313/99110557-3d42ca00-25f3-11eb-8feb-7c6d91a1bf54.png)

![изображение](https://user-images.githubusercontent.com/49781313/99110596-4df34000-25f3-11eb-963e-3b6d1c93975b.png)

**Comment by aandrey1993** [comment]
Страницы маршрутизации необязательно хранить в state в `redux`, так как они при работе приложения не должны меняться (нет смысла делать их реактивными). Для этого подойдет обычная константа. 

Чтобы избежать кучу проверок, дублирования кода, чтобы маршруты добавлялись в одном месте, при создания массива routes нужно сразу определить все необходимые поля. Небольшой пример:

```javascript
import React, { Suspense } from "react";
import { Switch, Route } from "react-router-dom";

const Home = React.lazy(() => import("../pages/Home"));
const Account = React.lazy(() => import("../pages/Account"));
const News = React.lazy(() => import("../pages/News"));

const routes = [
  {
    name: "home",
    path: "/",
    exact: true,
    component: Home
  },
  {
    name: "news",
    path: "/news",
    exact: false,
    component: News
  },
  {
    name: "account",
    path: "/account",
    exact: false,
    component: Account
  }
];

export default function MainRouter() {
  return (
    <Suspense fallback={<div>Загрузка ...</div>}>
      <Switch>
        {routes.map(route => (
          <Route
            key={route.name}
            path={route.path}
            exact={route.exact}
            component={route.component}
          />
        ))}
      </Switch>
    </Suspense>
  );
}
```

При необходимости,  в `routes ` можно добавить и иконку раздела, сделать export и использовать данную константу и для конструкции меню.

**Comment by Alexpol19** [comment]
Да, я думал убрать из redux маршруты. Но как-то не ставил целью и не  сделал
